kubernetes/helm/pinot/templates/controller/ingress.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.controller.ingress.enabled -}}
+{{- $ingressPath := .Values.controller.ingress.path -}}
+{{- $serviceName := include "pinot.controller.fullname" . -}}
+{{- $servicePort := .Values.controller.service.port -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ include "pinot.controller.fullname" . }}
+{{- if .Values.controller.ingress.annotations }}
+  annotations:
+{{ toYaml .Values.controller.ingress.annotations | indent 4 }}
+  {{- end }}
+  labels:
+    {{- include "pinot.controllerLabels" . | nindent 4 }}
+spec:
+{{- if .Values.controller.ingress.tls }}
+  tls:
+{{ toYaml .Values.controller.ingress.tls | indent 4 }}
+{{- end }}
+  rules:
+    {{- range .Values.controller.ingress.hosts }}
+    - host: {{ . }}
```

Comment:
this should also be configurable?

Reply:
Yes, and it is. It will load from the `values.yaml` file, you just need to overwrite it locally.

```yaml
  ingress:
    enabled: true
    annotations: {}
    tls: {}
    path: /
    hosts: 
       - host1
       - host2
```

The result from `helm template` command will look like this:
```yaml
# Source: pinot/templates/controller/ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: pinot-controller
  labels:
    helm.sh/chart: pinot-0.2.6-SNAPSHOT
    app: pinot
    chart: pinot-0.2.6-SNAPSHOT
    release: pinot
    app.kubernetes.io/version: "0.2.6-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
    heritage: Helm
    component: controller
spec:
  rules:
    - host: host1
      http:
        paths:
          - path: /
            backend:
              serviceName: pinot-controller
              servicePort: 9000
    - host: host2
      http:
        paths:
          - path: /
            backend:
              serviceName: pinot-controller
              servicePort: 9000
```

Reply:
👍